### PR TITLE
Relative paths in config

### LINKED ISSUE #137: Relative paths in the config should be appended to the config's parent
It's confusing for a path taken from the config to change depending on the current working directory. The only way to ensure predictability seems to start from the directory which contains the config.

### PR DESCRIPTION
**Pull request by EPashkin**
Fix #137

### CODE REVIEW

**Comment by EPashkin** [comment]
Rebased

File: src/pathutil.rs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        tp!(Some("/c"), "./a/b.txt", "/c/a/b.txt");
+        tp!(Some("/c"), "../a/b.txt", "/c/../a/b.txt");
+    }
+
+    #[test]
+    #[cfg(windows)]
+    fn test_make_base_on_windows() {
+        let none: Option<&str> = None;
+        tp!(none, "a/b.txt", "a/b.txt");
+        tp!(Some("/c"), "a/b.txt", "\\c\\a\\b.txt");
```
Comment (gkoz) [inline src/pathutil.rs]: `r"\c\a\b.txt"`

Comment (EPashkin) [inline src/pathutil.rs]: Oh, forgot

**Comment by EPashkin** [comment]
Fixed

**Comment by gkoz** [comment]
This feels too complicated for the task.

If `config_file` is absolute (will this mess up the error message BTW?) why should its parent be `Option`? Why make `path_based_on` (I'd call it `rebase`) handle the `Option`?

**Comment by EPashkin** [comment]
Don't though that I can safe unwrap parent of absolute path.

**Comment by EPashkin** [comment]
You prefer `config_file` stay unchanged and just extract path from this?

**Comment by gkoz** [comment]
I'm not a fan of making changing the relative path into absolute in the first place. If `parent` returns `None` if there's an empty parent ("Gir.toml") or an invalid input ("C:\" with no file). The first can be solved with prepending "." and the second by doing this after opening the file (and reading "C:\" will return an error).

**Comment by EPashkin** [comment]
It works only if we don't plan changing current folder.

**Comment by gkoz** [comment]
I believe codegen already assumes that we don't play with `current_dir`.

**Comment by EPashkin** [comment]
Then code really simply.

**Comment by EPashkin** [comment]
Strange, `PathBuf::from(".") == PathBuf::new()`, but seems works normally

**Comment by gkoz** [comment]
Oh, that's nice, `PathBuf::new()` is better than an explicit "`.`".

**Comment by EPashkin** [comment]
Done

**Comment by gkoz** [comment]
Looks like some lines over 100 chars now?

**Comment by EPashkin** [comment]
Yes, many. But too many tabs in `library_name` for example, that look bad anyway.

**Comment by EPashkin** [comment]
Fixed some

**Comment by gkoz** [comment]
Thanks! There seem to be opportunities to reducing boilerplate but perhaps some other time.
@homu r+

**Pull request merged**